### DISCUSSION TITLE
Configuration via JSON not working when running JAR file?

### DISCUSSION THREAD

**Discussion by hoylu-royberg** [klogging/klogging#260] [Q&A] [answered]
I'm using version 0.5.12 of Klogging in a service written in Kotlin and which uses Spring Boot, and I'm doing all the logging configuration via the following `klogging.json` file:
```json
{
  "sinks": {
    "stdout": {
      "renderWith": "RENDER_ECS",
      "sendTo": "STDOUT"
    }
  },
  "logging": [
    {
      "fromLoggerBase": "com.hoylu",
      "levelRanges": [
        {
          "fromMinLevel": "DEBUG",
          "toSinks": [
            "stdout"
          ]
        }
      ]
    }
  ],
  "kloggingMinLogLevel": "WARN"
}

```

This is working as expected when I run the service using e.g. `./gradlew bootRun`, but when I try to run the service using the compiled JAR file (`java -jar service.jar`), I don't get any log output at all. I've tried with various formatters, but still no log output. I have inspected the JAR file, and `klogging.json` is in the classpath (`BOOT-INF/classes/klogging.json` to be specific). I have tried setting `kloggingMinLogLevel` and `KLOGGING_MIN_LOG_LEVEL` to `TRACE`, but still nothing.

If I configure logging via DSL, then logging works as expected. Any ideas what the issue could be?

**Comment by mjstrasser** [comment]
Hi @hoylu-royberg thanks for your question.

I think I have seen this behaviour before but didn’t know how to address it. The solution is probably in specifying some extra information in a manifest, possibly in the Klogging library. I will look into it next week, when I have more time.

Perhaps if you find some information before I do, you can let me know.

**Comment by hoylu-royberg** [comment, marked as answer]
Hi Michael,

I added some print statements in `Files.jvm.kt` to get some more insight to what is going on, and it confirmed my suspicion; It's not finding any config files in the classpath.

I changed `readResourceText` slightly, and this seems to work for me:
```kotlin
internal fun readResourceText(resourcePath: String): String? =
    Thread.currentThread().contextClassLoader.getResourceAsStream(resourcePath)?.use { inputStream ->
        debug("Configuration", "Reading configuration from $resourcePath on the classpath")
        return inputStream.bufferedReader().use { it.readText() }
    }
```

**Comment by mjstrasser** [comment]
Hi @hoylu-royberg 

Thank you for figuring that out! When I wrote that code, I made a guess of which classloader to find, and my guess worked in my desktop testing. You can see from the comment that used to be in `Files.jvm.kt` that I wasn’t sure:

https://github.com/klogging/klogging/blob/d7c7fc17dd48270078ed8114a8a8019c6123e2f2/klogging/src/jvmMain/kotlin/io/klogging/config/Files.jvm.kt#L32-L40

I have made that change and released it in version 0.5.13, which should be on Maven Central soon.

**Reply by hoylu-royberg** [reply]
Awesome -- thank you! 😊
